README.md

```diff
@@ -1,69 +1,320 @@
+    //
+    // The split method here splits out the functionality of the GPIO Port B/C
+    // while taking a mutable borrow of an "enabler" that enables the clock for
+    // the port at the same time. The mutable borrow allows modification of the
+    // borrowed value while ensuring exclusive access.
+    //
+    let mut gpiob = device_periphs.GPIOB.split(&mut rcc.ahb);
+    let mut gpioc = device_periphs.GPIOC.split(&mut rcc.ahb);
+
+    // Configure pins PB6 as SCL and PB7 as SDA for I2C1.
+    //
+    let mut scl =
+        gpiob
+            .pb6
+            .into_af_open_drain(&mut gpiob.moder, &mut gpiob.otyper, &mut gpiob.afrl);
+    let mut sda =
+        gpiob
+            .pb7
+            .into_af_open_drain(&mut gpiob.moder, &mut gpiob.otyper, &mut gpiob.afrl);
+    scl.internal_pull_up(&mut gpiob.pupdr, true);
+    sda.internal_pull_up(&mut gpiob.pupdr, true);
 
-  // Create an instance of PaSpl.
-  //
-  let mut pa_spl = PaSpl::new(i2c);
+    // Create an instance of I2C1 with the pins.
+    //
+    let i2c = I2c::new(
+        device_periphs.I2C1,
+        (scl, sda),
+        100.kHz().try_into().unwrap(),
+        clocks,
+        &mut rcc.apb1,
+    );
 
-  // Get SPL value from the sensor.
-  //
-  let spl = pa_spl.get_latest_decibel().unwrap();
+    // Configure GPIO pins PC10 as TX and PC11 as RX for UART4.
+    //
+    let tx_pin = gpioc
+        .pc10
+        .into_af_push_pull(&mut gpioc.moder, &mut gpioc.otyper, &mut gpioc.afrh);
+    let rx_pin = gpioc
+        .pc11
+        .into_af_push_pull(&mut gpioc.moder, &mut gpioc.otyper, &mut gpioc.afrh);
+
+    // Create an instance of UART4 with the pins.
+    //
+    let mut uart4 = Serial::new(
+        device_periphs.UART4,
+        (tx_pin, rx_pin),
+        config::Config::default().baudrate(115_200.Bd()),
+        clocks,
+        &mut rcc.apb1,
+    );
+
+    // Use the I2C1 instance to create an instance of PaSpl.
+    //
+    let mut pa_spl = PaSpl::new(i2c);
+
+    // Create a buffer able to be converted to a string.
+    // 
+    let mut buffer: [u8; 8] = [0; 8];
+    let mut buf_writer = BufWriter::new(&mut buffer);
+
+    // Algo delay in milliseconds.
+    //
+    const ALGO_DELAY_MS: u16 = 500;
+
+    loop {
+        // Reset the buffer at the start of each iteration
+        //
+        buf_writer.reset();
+
+        // Get SPL value from the sensor.
+        //
+        let spl = pa_spl.get_latest_decibel().unwrap();
+
+        // Format string with SPL value, then covert to string.
+        //
+        write!(buf_writer, "SPL: {}\r", spl).unwrap();
+        let spl_str = buf_writer.as_str();
+
+        // Write the string out to the UART.
+        //
+        uart4.write_str(spl_str).unwrap_or_else(|_| {
+            loop {
+                // Failed to write to UART4.
+                asm::nop(); // If real app, replace with actual error handling.
+            }
+        });
+
+        // Limit algorithm to (1000 * (1 / UART_WRITE_DELAY_MS)) Hz.
```

Comment:
Fixed. Technically, the best practice for doxygen is to have a blank line, I believe, so using a blank line like this harmonizes the format of doc comments vs plain comments. I also think it improves the readability.

Reply:
I don't think that is a best practice for doxygen: https://www.doxygen.nl/manual/docblocks.html

It might wind up looking like that because the closing block comment is on its own line `*/` but you aren't using block comments.

Reply:
I dug around a bit (and asked our friend, ChatGPT): while it's not a hard rule anywhere, appending a blank line is generally viewed as a good practice for improving readability and making the format of comments consistent, so I'm going to stick to my guns on this one. I like the breathing room in the code when I'm reading through it.